src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -120,20 +107,20 @@ internal ITelemetryReceiver TelemetryReceiver
         /// Identifier of the component (libraries/SDK) consuming MSAL.NET.
         /// This will allow for disambiguation between MSAL usage by the app vs MSAL usage by component libraries.
         /// </summary>
-        public string Component { get; set; }
+        public string Component { get; set; }  // todo: have this only come in through config...
 
         /// <Summary>
         /// Gets the URL of the authority, or security token service (STS) from which MSAL.NET will acquire security tokens
         /// The return value of this property is either the value provided by the developer in the constructor of the application, or otherwise
         /// the value of the <see cref="DefaultAuthority"/> static member (that is <c>https://login.microsoftonline.com/common/</c>)
         /// </Summary>
-        public string Authority { get; }
+        public string Authority => ServiceBundle.Config.DefaultAuthorityInfo.CanonicalAuthority;
```

Comment:
This is the default if the user doesn't provide anything, but if they do set an authority, we should return it.

Reply:
No, DefaultAuthorityInfo returns the single AuthorityInfo class in the list of Authorities that has "isDefault=true" set on it.  We don't default to any authority on behalf of the developer, they have to tell us.

---
In reply to: [241516250](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/723#discussion_r241516250) [](ancestors = 241516250)